### PR TITLE
log information when fuse failed to open

### PR DESCRIPTION
**Pull request by benjaminhuanghuang**
utils/fuse: print more detailed error when mounting fusedev

Description: 
    when I started nydusd in a docker running in MacOS, nydusd existed without any helpful information left in log.

Reason: 
    MacOS dose not support fuse, that causes nydysd failed.

Fix: 
1. Log error message when FUSE failed to open
    
2. This fix cause Rustc 1.49 crash with error 
     " thread 'rustc' panicked at 'called `Option::unwrap()` on a `None` value'"

    So I upgrade the rust tool chain to 1.50
    
3. The Clippy report error:
     error: this function's return value is unnecessarily wrapped by `Result`
       --> utils/src/metrics.rs:411:5

     I fixed it accordingly.

Signed-off-by: Benjamin Huang <benjaminhuanghuang@gmail.com>

### CODE REVIEW

**Comment by liubogithub** [comment]
@benjaminhuanghuang 
Thanks for the fix.
DCO is not happy with missing SOB, can you please also put the above description and Signed-off-by in the commit log?
specifically, use `git commit --amend` and then make a force push `git push --force`

File: utils/src/fuse.rs [hunk 1]
```diff
@@ -252,12 +252,25 @@ fn fuse_kern_mount(
...
-    let meta = mountpoint.metadata()?;
+        .open(FUSE_DEVICE)
+    {
+        Ok(v) => v,
+        Err(e) => {
+            error!("FUSE failed to open. {}", e);
+            return Err(e);
+        }
+    };
+    let meta = match mountpoint.metadata() {
```
Comment (changweige) [inline utils/src/fuse.rs:267]: Um. I can't understand why such an update is causing `rustc` crashed.
But I suggest to remove all the existed target directories like `target-fusedev` and `target-virtiofs` from workspace. Then try to re-compile nydus by executing `make` from project home direcotry.
Sometimes, rustc is not stable. :-(

File: rust-toolchain [hunk 2]
```diff
@@ -1 +1 @@
-1.49.0
+1.50.0
```
Comment (changweige) [inline rust-toolchain:1]: If we truly have to upgrade toolchain. It is fine to me. But better that we can add a separated patch doing so.

File: utils/src/fuse.rs [hunk 3]
```diff
@@ -252,12 +252,25 @@ fn fuse_kern_mount(
...
         .create(false)
         .read(true)
         .write(true)
-        .open(FUSE_DEVICE)?;
-    let meta = mountpoint.metadata()?;
+        .open(FUSE_DEVICE)
+    {
+        Ok(v) => v,
+        Err(e) => {
+            error!("FUSE failed to open. {}", e);
```
Comment (changweige) [inline utils/src/fuse.rs:263]: How about just using error mapping. For example:
```
let file = OpenOptions::new()
        .create(false)
        .read(true)
        .write(true)
        .open(FUSE_DEVICE).map_err(|e| {error!("bla-bla");e})?
```
That is neater and clean

File: utils/src/metrics.rs [hunk 4]
```diff
@@ -484,7 +484,7 @@ pub fn export_files_stats(
             if !latest_read_files {
                 v.export_files_stats()
             } else {
-                v.export_latest_read_files()
+                Ok(v.export_latest_read_files())
```
Comment (changweige) [inline utils/src/metrics.rs:487]: Clippy of version 1.51 or higher does not complain about this anymore.

**Review by changweige — CHANGES_REQUESTED** [review]

**Comment by changweige** [comment]
@benjaminhuanghuang  Can you add a tittle line to your patch commit log?
For example: utils/fuse:  print more detailed error when mounting fusedev

Description:
    ....

**Comment by imeoer** [comment]
@benjaminhuanghuang Thanks for this patch. I noticed that the [build](https://github.com/dragonflyoss/image-service/blob/master/misc/musl-static/Dockerfile) and [smoke test](https://github.com/dragonflyoss/image-service/blob/master/misc/nydus-smoke/Dockerfile) scripts still depend on 1.49.0, could you help change them together?

File: utils/src/fuse.rs (hunk 1 above)
Comment (bhuang-rms) [reply utils/src/fuse.rs:267]: I built my patch in two different environment. rustc crashes in both of them.

File: rust-toolchain (hunk 2 above)
Comment (bhuang-rms) [reply rust-toolchain:1]: Without upgrading the toolchain, this path can not pass compiling , is that acceptable?

File: utils/src/fuse.rs (hunk 3 above)
Comment (bhuang-rms) [reply utils/src/fuse.rs:263]: yes, I agree

**Comment by bhuang-rms** [comment]
> @benjaminhuanghuang Can you add a tittle line to your patch commit log?
> For example: utils/fuse: print more detailed error when mounting fusedev
> 
> Description:
> ....

The PR has a title. Because a PR is always composed by several commits. It is better to put the  summary text to the title of the PR.

File: utils/src/fuse.rs (hunk 3 above)
Comment (benjaminhuanghuang) [reply utils/src/fuse.rs:261]: Yes, I agree

File: rust-toolchain (hunk 2 above)
Comment (benjaminhuanghuang) [reply rust-toolchain:1]: Agree

File: utils/src/metrics.rs (hunk 4 above)
Comment (liubogithub) [reply utils/src/metrics.rs:487]: 1.51.0 reports this error too, and even more Clippy errors, I think we can let this go and fix the others in a separate patch later.

Comment (changweige) [reply utils/src/metrics.rs:487]: True.

**Review by changweige — APPROVED** [review]
Looks good

**Pull request merged**